README.md

```diff
@@ -1,26 +1,72 @@
+It is a critical component for the "reactive" mode of the GitHub Runner Charm.
 
-> **Warning**
-Work in progress: This charm is still in development and not yet ready for use in production.
 
-The github-runner-webhook-router charm is a Juju charm that provides a webhook router for GitHub Actions self-hosted runners. 
-This charm is designed to be used in conjunction with the [github-runner-operator](https://github.com/canonical/github-runner-operator) charm.
 
 
-## Other resources
+Like any Juju charm, this charm supports one-line deployment, configuration, integration, scaling, and more.
+For the GitHub Runner Webhook Router, this includes
 
-<!-- If your charm is documented somewhere else other than Charmhub, provide a link separately. -->
+* All the general and Flask-related features of [paas-charm](https://github.com/canonical/paas-charm)
+* Configuring the routing table
+* Configuring a webhook secret to improve the security of the webhook endpoint
 
-- [Contributing](CONTRIBUTING.md) <!-- or link to other contribution documentation -->
+For information on how to deploy, integrate and manage this charm, please see the official
+[GitHub Runner Webhook Router Documentation](https://charmhub.io/github-runner-webhook-router).
 
-- See the [Juju SDK documentation](https://juju.is/docs/sdk) for more information about developing and improving charms.
+
+## Get started
+As the charm is designed to be used in conjunction with the [GitHub Runner](https://github.com/canonical/github-runner-operator) charm,
+please see [How to set up reactive spawning](https://charmhub.io/github-runner/docs/how-to-reactive) to learn how to deploy the charm.
+
+The charm requires an integration with mongodb (either the machine or k8s charm), otherwise it will
+go into a blocked state.
+
+### Basic operations
+
+Configure a routing table to decide which labels should be routed to which GitHub Runner charm application:
+
+```shell
+cat <<EOF > routing_table.yaml
+- large: [large, x64]
+- large-arm: [large, arm64]
+- small: [small]
+EOF
+juju config github-runner-webhook-router routing-table="$(cat routing_table.yaml)"
+```
+
+Change the default flavor to be used for jobs only containing default self-hosted-labels:
+
+```shell
+juju config github-runner-webhook-router default-flavour=small
+```
+
+Change the webhook secret used for webhook validation:
+
+```shell
+juju config github-runner-webhook-router webhook-secret=<your-secret>
+```
+
+
+## Learn more
+* [Read more](https://charmhub.io/github-runner-webhook-router)
+* [Developer documentation](https://github.com/canonical/github-runner-webhook-router/blob/main/CONTRIBUTING.md)
+* [PaaS Charm](https://github.com/canonical/paas-charm)
+* [Write your first Kubernetes charm for a Flask app](https://juju.is/docs/sdk/write-your-first-kubernetes-charm-for-a-flask-app)
+* [How to build a 12-Factor app charm](https://juju.is/docs/sdk/build-a-12-factor-app-charm)
+
```

Comment:
I would include another item in this list, `Troubleshooting`, and include a link to the 12-factor Matrix channel, `https://matrix.to/#/#12-factor-charms:ubuntu.com`